scripts/yggdrasil/install

```diff
@@ -43,19 +43,33 @@ fi
 sudo sed -i "s/IfName: auto/IfName: ygg0/" /etc/yggdrasil.conf 
 
 # Change port to 12345 for firewall
-YGGDRASIL_PORT=$(sudo cat /etc/yggdrasil.conf | grep \ Listen: | awk  '{print $2}' | tr -d \")
-YGGDRASIL_PORT=$(printf "%q" "$YGGDRASIL_PORT") # Escape for sed
-sudo sed  -i "s/$YGGDRASIL_PORT/\[::\]:12345/" /etc/yggdrasil.conf
+sudo sed -i "s/Listen: \[\]/Listen\: [\"tcp:\/\/\[::\]:12345\"\]/" /etc/yggdrasil.conf
 
+sudo systemctl enable yggdrasil.service
+sudo systemctl start yggdrasil.service 
 
-# Set node name
-if [ -z "${NEWHOSTNAME}" ]; then 
-    NEWHOSTNAME=$(cat /etc/hostname)
+# Dont count yggdrasill ips as internet ip addresses
+if [ -f "/etc/gai.conf" ]; then
+  sudo sed -i '/label 200::/d' /etc/gai.conf
+  sudo sed -i '/label 300::/d' /etc/gai.conf
+  echo "label 200::/7 6" | sudo tee --append /etc/gai.conf
+  echo "label 300::/7 6" | sudo tee --append /etc/gai.conf
 fi
-sudo sed -i "\$i\ \ NodeInfo:\n  {\n   name: ${NEWHOSTNAME}\n  }\n"  /etc/yggdrasil.conf
 
-sudo systemctl enable yggdrasil.service
-sudo systemctl start yggdrasil.service 
 
 # Allow nginx access to yggdrasil socket when installed
 sudo usermod -a -G yggdrasil www-data || true
+
+# Define new hostname
+if [ -z "${NEWHOSTNAME}" ]; then
+   NEWHOSTNAME=$(sudo ifconfig ygg0 | grep -E 'inet6 2[0-9a-fA-F]{2}:' | awk '{print $2}' | sed 's/[",]//g' | sed "s/.*:/$MESH_NAME-Y/g")
+   export NEWHOSTNAME
+fi
+
+# Set node name
+if [ -z "${NEWHOSTNAME}" ]; then 
+    NEWHOSTNAME=$(cat /etc/hostname)
+fi
+
+sudo sed -i "/NodeInfo: {}/d" /etc/yggdrasil.conf
+sudo sed -i "\$i\ \ NodeInfo:\n  {\n   name: ${NEWHOSTNAME}\n  }\n"  /etc/yggdrasil.conf
```

Comment:
This should also be updated in `nodeinfo.json`. Also, if the hostname has changed, shouldn't it be changed in `/etc/hostname`, or does that already happen?

Reply:
hostname ONLY changed during install.

This is NOT where the hostname gets set. What your thinking about is 

https://github.com/tomeshnet/prototype-cjdns-pi/blob/ygg-tunnel-ipv6/scripts/install2#L407

Reply:
https://github.com/tomeshnet/prototype-cjdns-pi/issues/372

Reply:
Okay. What will happen if someone tries to install Yggdrasil with this script later, and so `NEWHOSTNAME` isn't set?

Also, about that link and any other places when the hostname is set, why not use the `hostname` command, **as well as** changing the file? Like this: `sudo hostname NAME`